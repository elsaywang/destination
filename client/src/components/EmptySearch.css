:local(.noResultMessage) {
    margin: -0.75em 0 0 1.85em;
}
:local(.exploreMessage) {
    margin: -0.75em 0 0 1.5em;
}
:local(.noResultImage) {
    width: 90px;
    height: 80px;
    margin-left: -8px;
}
:local(.exploreImage) {
    width: 190px;
    height: 101px;
}
