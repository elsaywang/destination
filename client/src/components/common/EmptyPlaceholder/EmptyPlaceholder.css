:local(.title) {
    padding-left: 1.5em;
}
:local(.message) {
    font-style: italic;
    color: var(--spectrum-global-color-grey500);
}
