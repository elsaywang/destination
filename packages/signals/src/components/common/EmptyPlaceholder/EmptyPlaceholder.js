import React from 'react';
import PropTypes from 'prop-types';
import styles from './EmptyPlaceholder.css';
import classNames from 'classnames';
import Heading from '@react/react-spectrum/Heading';

function EmptyPlaceholder(props) {
    return (
        <div className={props.className} data-test="empty">
            {props.children}
            <Heading size={2} className={styles.title}>
                {props.title}
            </Heading>
            <div className={classNames(styles['message'], props.messageStyle)}>{props.message}</div>
        </div>
    );
}

EmptyPlaceholder.propTypes = {
    title: PropTypes.string.isRequired,
    message: PropTypes.string.isRequired,
};

export default EmptyPlaceholder;
