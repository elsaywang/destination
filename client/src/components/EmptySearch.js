import React from 'react';
import PropTypes from 'prop-types';
import EmptyPlaceholder from './common/EmptyPlaceholder';
import styles from './EmptySearch.css';
import Explore from '../images/explore.svg';
import NoResult from '../images/noResult.svg';
import { getEmptyOptions } from '../constants/emptyOptions';
import { getStaticUrl } from '../lib/getStaticUrl';

function EmptySearch({ variant, className }) {
    const getMessageStyle = () => {
        switch (variant) {
            case 'explore':
                return styles.exploreMessage;
            case 'noResult':
                return styles.noResultMessage;
            case 'errorFetching':
                return styles.noResultMessage;
            default:
                return '';
        }
    };
    const getImageClass = () => {
        switch (variant) {
            case 'explore':
                return styles.exploreImage;
            case 'noResult':
                return styles.noResultImage;
            case 'errorFetching':
                return styles.noResultImage;
            default:
                return '';
        }
    };
    const getImageSrc = () => {
        switch (variant) {
            case 'explore':
                return `${getStaticUrl()}${Explore}`;
            case 'noResult':
                return `${getStaticUrl()}${NoResult}`;
            case 'errorFetching':
                return `${getStaticUrl()}${NoResult}`;
            default:
                return '';
        }
    };

    return (
        <EmptyPlaceholder
            className={className}
            title={getEmptyOptions(variant).title}
            message={getEmptyOptions(variant).message}
            messageStyle={getMessageStyle()}>
            <img
                src={getImageSrc()}
                data-test={getEmptyOptions(variant).dataTest}
                className={getImageClass()}
                alt={getEmptyOptions(variant).imageAlt}
            />
        </EmptyPlaceholder>
    );
}

EmptySearch.propTypes = {
    variant: PropTypes.oneOf(['explore', 'noResult']).isRequired,
};

export default EmptySearch;
