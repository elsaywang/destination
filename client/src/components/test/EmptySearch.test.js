import React from 'react';
import { shallow } from 'enzyme';
import EmptySearch from '../EmptySearch';
import EmptyPlaceholder from '../../components/common/EmptyPlaceholder';
import Explore from '../../images/explore.svg';
import NoResult from '../../images/noResult.svg';

describe('<EmptySearch/> component', () => {
    describe('when variant is `explore` passes in props', () => {
        const wrapper = shallow(<EmptySearch variant="explore" />);

        it('matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('renders <EmptyPlaceholder/> component with Explore Image', () => {
            expect(wrapper.find(EmptyPlaceholder).exists()).toBe(true);
            expect(wrapper.find('img').props().src).toEqual(Explore);
        });
    });
    describe('when variant is `noResult` passes in props', () => {
        const wrapper = shallow(<EmptySearch variant="noResult" />);

        it('matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('renders <EmptyPlaceholder/> component with NoResult Image', () => {
            expect(wrapper.find(EmptyPlaceholder).exists()).toBe(true);
            expect(wrapper.find('img').props().src).toEqual(NoResult);
        });
    });
});
