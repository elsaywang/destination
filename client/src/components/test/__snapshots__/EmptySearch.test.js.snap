// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptySearch/> component when variant is \`explore\` passes in props matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EmptySearch
    variant="explore"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <img
        alt="Start exploring. Search by key-value pairs or use only filters to get insights on signals."
        className={undefined}
        data-test="start-exploring"
        src="//localhost:3000/explore.svg"
      />,
      "className": undefined,
      "message": "Search by key-value pairs or use only filters to get insights on signals.",
      "messageStyle": undefined,
      "title": "Start exploring.",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "Start exploring. Search by key-value pairs or use only filters to get insights on signals.",
        "className": undefined,
        "data-test": "start-exploring",
        "src": "//localhost:3000/explore.svg",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <img
          alt="Start exploring. Search by key-value pairs or use only filters to get insights on signals."
          className={undefined}
          data-test="start-exploring"
          src="//localhost:3000/explore.svg"
        />,
        "className": undefined,
        "message": "Search by key-value pairs or use only filters to get insights on signals.",
        "messageStyle": undefined,
        "title": "Start exploring.",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Start exploring. Search by key-value pairs or use only filters to get insights on signals.",
          "className": undefined,
          "data-test": "start-exploring",
          "src": "//localhost:3000/explore.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<EmptySearch/> component when variant is \`noResult\` passes in props matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EmptySearch
    variant="noResult"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <img
        alt="No results found. Refine the search query and try searching again."
        className={undefined}
        data-test="no-result-found"
        src="//localhost:3000/noResult.svg"
      />,
      "className": undefined,
      "message": "Refine the search query and try searching again.",
      "messageStyle": undefined,
      "title": "No results found.",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "No results found. Refine the search query and try searching again.",
        "className": undefined,
        "data-test": "no-result-found",
        "src": "//localhost:3000/noResult.svg",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <img
          alt="No results found. Refine the search query and try searching again."
          className={undefined}
          data-test="no-result-found"
          src="//localhost:3000/noResult.svg"
        />,
        "className": undefined,
        "message": "Refine the search query and try searching again.",
        "messageStyle": undefined,
        "title": "No results found.",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "No results found. Refine the search query and try searching again.",
          "className": undefined,
          "data-test": "no-result-found",
          "src": "//localhost:3000/noResult.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
